htsget-search/src/htsget/cram_search.rs

```diff
@@ -183,92 +135,88 @@ impl<'a, S> CramSearch<'a, S>
     seq_start: Option<i32>,
     seq_end: Option<i32>,
     crai_index: &[crai::Record],
-    cram_reader: &mut noodles_cram::Reader<BufReader<File>>,
+    cram_reader: &mut noodles_cram::Reader<File>,
     predicate: F,
   ) -> Result<Vec<BytesRange>>
-    where F: Fn(&Record) -> bool
+  where
+    F: Fn(&Record) -> bool,
   {
     // This could be improved by using some sort of index mapping.
-    let mut byte_ranges: Vec<BytesRange> = crai_index.iter().zip(crai_index.iter().skip(1))
+    let mut byte_ranges: Vec<BytesRange> = crai_index
+      .iter()
+      .zip(crai_index.iter().skip(1))
       .filter_map(|(record, next)| {
         if predicate(record) {
           Self::bytes_ranges_for_record(ref_seq, seq_start, seq_end, record, next)
         } else {
           None
         }
-      }).collect();
+      })
+      .collect();
 
-    let last = crai_index.last().ok_or_else(|| HtsGetError::invalid_input("No entries in CRAI"))?;
+    let last = crai_index
+      .last()
+      .ok_or_else(|| HtsGetError::invalid_input("No entries in CRAI"))?;
     if predicate(last) {
       // An implementation based on file size might be better.
```

Comment:
This could be also solved now, right? The `head` method is already in the storage trait